src/Composer/ComposerWindow.vala

```diff
@@ -37,20 +37,25 @@ public class Mail.ComposerWindow : Hdy.ApplicationWindow {
     }
 
     construct {
-        composer_widget.discarded.connect (() => {
-            close ();
-        });
-        composer_widget.sent.connect (() => {
-            close ();
-        });
-
         var titlebar = new Hdy.HeaderBar () {
             has_subtitle = false,
             show_close_button = true,
             title = _("New Message")
         };
         titlebar.get_style_context ().add_class ("default-decoration");
 
+        composer_widget.discarded.connect (() => {
+            close ();
+        });
+        composer_widget.sent.connect (() => {
+            close ();
+        });
+        composer_widget.updated_subject.connect ((subject) => {
+            title = subject ?? _("New Message");
+            title = title.length > 0 ? title : _("New Message");
+            titlebar.title = title;
```

Comment:
Pardon my ignorance, I very new regarding `Vala` language, I find it hard to think of a scenario where subject could be `null` after being changed by the user. In any case, perhaps we could DRY off the `_("New Message")`, with something like:
```
    bool empty_subject = subject == null || subject.length == 0;                                                                                                               
    titlebar.title = empty_subject ? _("New Message") : subject;
```